### ISSUE TITLE
Instant Access DD Snapshot - Pv2 and Ultra Disk Snapshot Instantly Available for restoring disks upon creation

### ISSUE THREAD

**Issue by AdamZLi** [Azure/azure-cli#30007]
### Preconditions

- [X] No need to upgrade Python SDK or the Python SDK is ready.

### Related command

az snapshot create
az snapshot list
az snapshot show
az disk create
az disk list
az disk show

### Resource Provider

Microsoft.Compute/disks

### Description of Feature or Work Requested

_Note: CLI team should be able to pick up the work by end of Nov 2024_

Customers want to gain instant access to snapshot taken from Premium SSD v2 and Ultra disks so that customers can restore disks from snapshots as soon as snapshots are created.

### Minimum API Version Required

N/A

### Swagger PR link / SDK link

N/A

### Target Date

2024-12-31

### PM Contact

adli@microsoft.com

### Engineer Contact

akashpargat@microsoft.com

### Additional context

Backend changes is expected to be ready by end of Nov for CLI team to consume and start the work.

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by AdamZLi** [comment]
Feature tracked in ADO - [Feature 24923395](https://msazure.visualstudio.com/One/_workitems/edit/24923395): [CLI] Instant Access of DD snapshots for disk creation.

**Comment by zhoxing-ms** [comment]
@AdamZLi May I ask if this feature request is related to security? Please note that currently, due to the CLI team being busy with the security wave and prioritizing security in this semester, so all the feature requests unrelated to security will be postponed. cc PM @Jacekey23

**Comment by AdamZLi** [comment]
++ Msft email.

Hi Xing, What's the timeline for CLI to make up non-security related work?

Best,
Adam

**Comment by zhoxing-ms** [comment]
@AdamZLi Since security is still the no.1 top things to catch up within Azure. but CLI will be in continous planning manner. If things can be justify as highly customer impact. please send the justification and we'll take it into consideration.

**Comment by yanzhudd** [comment]
Hi @AdamZLi 
please specify the impacted parameters of Azure CLI and the corresponding Swagger link

**Comment by AdamZLi** [comment]
Hi Zhoxing, Yanzhu,

Apologize for the late response.

As mentioned in the GitHub thread, sharing the justification here: Instant Access is requested by multiple critical customers, including MongoDB, SAP, Black Rock, Epic, Electronic Art, Unity, and is marked as a critical blocker by SAP, Epic and BLK for adopting our latest disk storage SKU - Premium SSD v2. We have customer commitment to deliver the feature by Q2 CY2025.

As for "impacted parameters of Azure CLI and the corresponding Swagger link" - @***@***.***>, is this something you can help with?

Best,
Adam

From: Adam Li ***@***.***>
Sent: Wednesday, November 6, 2024 2:01 PM
To: Azure/azure-cli ***@***.***>
Cc: Azure/azure-cli ***@***.***>; Mention ***@***.***>; Adam Li ***@***.***>
Subject: [EXTERNAL] Re: [Azure/azure-cli] Instant Access DD Snapshot - Pv2 and Ultra Disk Snapshot Instantly Available for restoring disks upon creation (Issue #30007)

Hi Yan and Xing, sorry for the late response. I don't check my gmail often enough, adding my msft email.

Will provide a response using my msft email shortly.

Best,
Adam

On Tue, 22 Oct 2024 at 02:36, Yan Zhu ***@***.******@***.***>> wrote:

Hi @AdamZLi<https://github.com/AdamZLi>
please specify the impacted parameters of Azure CLI and the corresponding Swagger link

-

**Comment by yanzhudd** [comment]
> As for "impacted parameters of Azure CLI and the corresponding Swagger link" - @***@***.***>, is this something you can help with?

Hi @AdamZLi 
although we can relate CLI parameter to the backend argument from CLI perspective, you need to provide us with specific impacted backend arguments, otherwise we don't understand the backend args, and there's no way for us to develop CLI commands.

**Comment by yanzhudd** [comment]
Hi @AdamZLi 
please provide the above information ASAP if you want to unblock this feature.
